CHANGELOG.md

```diff
@@ -23,3 +23,6 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 ### Security
 
 [Unreleased 3.x]: https://github.com/opensearch-project/OpenSearch/compare/3.1...main
+### Added
+[Derive Source] Adding back "Adding integration of derived source feature across diff paths" with fixes for IT #18565
```

Comment:
Nitpick...but the fact that it is "adding back" is not important to a release notes reader because the original commits were never released. Just phrase this as what this commit does.

Also, even more pedantic of a nitpick, please use the imperative "Add integration..." instead of "Adding integration...".